rest/doc_api.go

```diff
@@ -393,7 +397,10 @@ func (h *handler) handlePutDoc() error {
 		bodyRev := body[db.BodyRev]
 		if oldRev := h.getQuery("rev"); oldRev != "" {
 			body[db.BodyRev] = oldRev
-		} else if ifMatch := h.rq.Header.Get("If-Match"); ifMatch != "" {
+		} else if ifMatch, err := h.getEtag("If-Match"); ifMatch != "" {
+			if err != nil {
```

Comment:
`getEtag` always returns an `""` value with the error, so this error handling would never be hit.

In this particular case, given it's a rev ID, we know it should have a specific format to match, so SG is going to fail further down if the provided value is quoted or otherwise wrong anyway, so I think we can just ignore the error.